slo_generator/backends/datadog.py

```diff
@@ -125,6 +125,30 @@ def query_slo(self, timestamp, window, slo_config):
         bad_event_count = valid_event_count - good_event_count
         return (good_event_count, bad_event_count)
 
+    def query_slo_monitor(self, timestamp, window, slo_config):
```

Comment:
To avoid code duplication we can combine the two methods `query_slo_monitor` and `query_slo_metric` into the original name and just retrieve what we need based on the SLO type.

e.g:

```
data = self.client.ServiceLevelObjective.history(id=slo_id, from_ts=from_ts, to_ts=timestamp)
try: # metric-based SLI
  good_event_count = data['data']['series']['numerator']['sum']
  valid_event_count = data['data']['series']['denominator']['sum']
  bad_event_count = valid_event_count - good_event_count
  return (good_event_count, bad_event_count)
except IndexError: # monitor-based SLI
  sli_value = data['data']['data']['overall']['sli_value'] / 100
  return sli_value
```

Reply:
Great idea !